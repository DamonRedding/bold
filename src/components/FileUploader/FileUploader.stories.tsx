import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import React from 'react'

import { FileUploader } from './FileUploader'

storiesOf('Components|FileUploader', module)
  .add('initial', () => (
    <FileUploader
      text='Click or drop file here'
      onDrop={action('onDrop')}
      onDropAccepted={action('onDropAccepted')}
      onDropRejected={action('onDropRejected')}
      onFileDialogCancel={action('onFileDialogCancel')}
      onDragOver={action('onDragOver')}
      onDragEnter={action('onDragEnter')}
      onDragLeave={action('onDragLeave')}
      disabled
    />
  ))
  .add('uploading', () => (
    <FileUploader
      text='Click or drop file here'
      file={{
        progress: 40,
        selectedFile: new File([], 'file_test.pdf', { type: 'text/pdf' }),
        uploading: true,
      }}
      onDrop={action('onDrop')}
      onDropAccepted={action('onDropAccepted')}
      onDropRejected={action('onDropRejected')}
      onFileDialogCancel={action('onFileDialogCancel')}
      onDragOver={action('onDragOver')}
      onDragEnter={action('onDragEnter')}
      onDragLeave={action('onDragLeave')}
    />
  ))
  .add('completed', () => (
    <FileUploader
      text='Click or drop file here'
      file={{
        selectedFile: new File([], 'file_test.pdf', { type: 'text/pdf' }),
      }}
      onDrop={action('onDrop')}
      onDropAccepted={action('onDropAccepted')}
      onDropRejected={action('onDropRejected')}
      onFileDialogCancel={action('onFileDialogCancel')}
      onDragOver={action('onDragOver')}
      onDragEnter={action('onDragEnter')}
      onDragLeave={action('onDragLeave')}
    />
  ))
