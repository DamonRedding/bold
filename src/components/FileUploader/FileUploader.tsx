import React, { CSSProperties } from 'react'
import { DropzoneOptions, useDropzone } from 'react-dropzone'

import { focusBoxShadow, Theme, useStyles } from '../../styles'
import { format } from '../../util/byte'
import { HFlow } from '../HFlow'
import { Icon } from '../Icon/Icon'
import { Progress } from '../Progress/Progress'
import { Text } from '../Text'

export interface FileUploaderProps extends DropzoneOptions {
  file?: FileProps
  text?: string
}

export interface FileProps {
  error?: boolean
  progress?: number
  selectedFile?: File
  uploading?: boolean
}

export function FileUploader(props: FileUploaderProps) {
  const { file, text, disabled, ...rest } = props
  const { classes, css } = useStyles(createStyles)
  const { getRootProps, getInputProps, isDragActive, isDragAccept } = useDropzone(rest)

  return (
    <div
      className={css(
        classes.componentWrapper,
        isDragActive && classes.dragBackground,
        disabled && classes.pointerEventsNone
      )}
      {...getRootProps()}
    >
      <input {...getInputProps()} />
      <div className={css(classes.buttonWrapper, disabled && classes.disabled)}>
        <div className={css(classes.dropzone, isDragActive && classes.dragActive)}>
          {isDragAccept}
          <HFlow justifyContent='center' alignItems='center' hSpacing={0.5}>
            <Icon fill={disabled ? 'disabled' : 'secondary'} icon='upload' />
            <Text color={disabled ? 'disabled' : 'secondary'} fontSize={0.875} fontWeight='bold'>
              {text}
            </Text>
          </HFlow>
        </div>
      </div>

      {file && <FileDetails file={file} />}
    </div>
  )
}

export interface FileDetailsProps {
  file: FileProps
}

export function FileDetails(props: FileDetailsProps) {
  const { file } = props
  const { css, classes } = useStyles(createStyles)

  const returnExtension = () => {
    const { type } = file.selectedFile
    const typeSplit = type.split('/')
    if (typeSplit.length > 1) {
      return typeSplit[1]
    } else if (!typeSplit[0]) {
      return null
    } else {
      return typeSplit[0]
    }
  }

  return (
    <div className={css(classes.file, classes.fileDetailsContainer)}>
      <FileExtension extension={returnExtension()} />
      <div className={classes.fileDetails}>
        <FileInfo file={file} />
        {file.uploading && <Progress value={file.progress} />}
      </div>
    </div>
  )
}

export interface FileInfoProps {
  file: FileProps
}

export function FileInfo(props: FileInfoProps) {
  const { file } = props
  const { classes } = useStyles(createStyles)

  return (
    <div className={classes.fileDetailsContainer}>
      {!file.error && !file.uploading && (
        <Icon icon='checkDefault' fill='primary' size={1} style={{ marginRight: 5 }} />
      )}
      <Text fontWeight='bold' style={{ marginRight: 10 }}>
        {file.selectedFile.name}
      </Text>
      <Text>{format(file.selectedFile.size, 0)}</Text>
    </div>
  )
}

export interface FileExtensionProps {
  extension: string
}

export function FileExtension(props: FileExtensionProps) {
  const { extension } = props
  const { classes } = useStyles((theme: Theme) => ({
    root: {
      backgroundColor: theme.pallete.surface.background,
      border: `1px solid ${theme.pallete.divider}`,
      padding: '1rem',
    },
  }))

  return (
    <Text color='secondary' style={classes.root} fontWeight='bold'>
      {extension}
    </Text>
  )
}

export const createStyles = (theme: Theme) => ({
  componentWrapper: {
    backgroundColor: theme.pallete.surface.main,
    transition: 'box-shadow .2s ease',

    '&:focus': {
      outline: 'none',
      boxShadow: focusBoxShadow(theme),
    },
  } as CSSProperties,
  buttonWrapper: {
    border: `1px solid` + theme.pallete.gray.c40,
    borderRadius: theme.radius.popper,
    cursor: 'pointer',
    padding: '0.25em',
  } as CSSProperties,
  dropzone: {
    borderRadius: theme.radius.popper,
    padding: '0.52em',
  } as CSSProperties,
  dragBackground: {
    backgroundColor: theme.pallete.surface.background,
  } as CSSProperties,
  dragActive: {
    border: `1px dashed` + theme.pallete.gray.c40,
  } as CSSProperties,
  pointerEventsNone: {
    pointerEvents: 'none',
  } as CSSProperties,
  disabled: {
    border: `1px solid` + theme.pallete.gray.c80,
  } as CSSProperties,

  file: {
    borderRadius: theme.radius.popper,
    borderBottom: '1px solid ' + theme.pallete.gray.c80,
    borderLeft: '1px solid ' + theme.pallete.gray.c80,
    borderRight: '1px solid ' + theme.pallete.gray.c80,
    padding: '1rem',
  },
  fileDetailsContainer: {
    alignItems: 'center',
    display: 'flex',
  },
  fileDetails: {
    flexGrow: 1,
    marginLeft: '1rem',
  },
})
