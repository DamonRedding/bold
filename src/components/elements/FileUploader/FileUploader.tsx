import { css as emotionCss } from 'emotion'
import * as React from 'react'
import Dropzone, * as DropzoneAll from 'react-dropzone'

import { Styles, Theme, withStyles, WithStylesProps } from '../../../styles'
import { format } from '../../../util/byte'
import { HFlow } from '../../layout/Flow/HFlow'
import { Icon } from '../Icon/Icon'
import { ProgressIndicator } from '../ProgressIndicator/ProgressIndicator'
import { Text } from '../textual/Text/Text'

export interface FileUploaderProps extends WithStylesProps {
    accept?: string
    file?: FileProps
    maxSize?: number
    text?: string
    onUpload?(selectedFile: any): void
}

export interface FileProps {
    error?: boolean
    progress?: number
    selectedFile?: any
    uploading?: boolean
}

const DropzoneCmp: any = Dropzone || DropzoneAll

@withStyles
export class FileUploader extends React.Component<FileUploaderProps> {

    uploadImage = (file) => {
        if (file && file.length > 0) {
            const selectedFile = file[0]

            this.props.onUpload && this.props.onUpload(selectedFile)
        }
    }

    render() {
        const {
            accept,
            css,
            maxSize,
            theme,
        } = this.props

        const styles: Styles = {
            dropzone: {
                cursor: 'pointer',
                padding: '0.25rem',
                '& .accept-dashed-border': {
                    alignItems: 'center',
                    display: 'flex',
                    flexDirection: 'column',
                    border: '1px solid transparent',
                    borderStyle: 'dashed',
                    paddingBottom: '0.5rem',
                    paddingTop: '0.5rem',
                },
            },
            dropzoneAccept: {
                backgroundColor: theme.pallete.surface.background,
                '& .accept-dashed-border': {
                    borderColor: theme.pallete.divider + '!important',
                },
            },
            file: {
                borderTop: '1px solid ' + theme.pallete.divider,
                padding: '1rem',
            },
            wrapper: {
                backgroundColor: theme.pallete.surface.main,
                borderRadius: theme.radius.paper,
                border: '1px solid ' + theme.pallete.divider,
            },
        }

        return (
            <div className={css(styles.wrapper)}>
                <DropzoneCmp
                    maxSize={maxSize}
                    accept={accept}
                    className={emotionCss(styles.dropzone)}
                    acceptClassName={emotionCss(styles.dropzoneAccept)}
                    onDrop={this.uploadImage}
                    multiple={false}
                >
                    <div className='accept-dashed-border'>
                        <HFlow alignItems='center' hSpacing={0.5}>
                            <Icon color='gray40' icon='upload' />
                            <Text color='gray40' size={0.875} weight='bold'>{this.props.text}</Text>
                        </HFlow>
                    </div>
                </DropzoneCmp>
                {this.props.file &&
                    <FileDetails
                        file={this.props.file}
                        theme={theme}
                    />
                }
            </div>
        )
    }

}

interface FileDetailsProps {
    file: FileProps
    theme: Theme
}

class FileDetails extends React.Component<FileDetailsProps> {

    render() {
        const styles: Styles = {
            info: {
                flexGrow: 1,
                marginLeft: '1rem',
            },
            wrapper: {
                alignItems: 'center',
                borderTop: '1px solid ' + this.props.theme.pallete.divider,
                display: 'flex',
                padding: '1rem',
            },
        }

        return (
            <div className={emotionCss(styles.wrapper)}>
                <FileExtension extension='PDF' theme={this.props.theme} />
                <div className={emotionCss(styles.info)}>
                    <FileInfo file={this.props.file} theme={this.props.theme} />
                    {this.props.file.uploading &&
                        <ProgressIndicator value={this.props.file.progress} />
                    }
                </div>
            </div>
        )
    }

}

interface FileInfoProps {
    file: FileProps
    theme: Theme
}

class FileInfo extends React.PureComponent<FileInfoProps> {

    render() {
        const styles = {
            wrapper: {
                alignItems: 'center',
                display: 'flex',
            },
        }

        return (
            <div className={emotionCss(styles.wrapper)}>
                {!this.props.file.error && !this.props.file.uploading &&
                    <Icon icon='check' color='primary' size={1} style={{ marginRight: 5 }} />
                }
                <Text color='gray30' size={0.75} weight='bold' style={{ marginRight: 10 }}>
                    {this.props.file.selectedFile.name}
                </Text>
                <Text size={0.75}>{format(this.props.file.selectedFile.size, 0)}</Text>
            </div>
        )
    }

}

interface FileExtensionProps {
    extension: string
    theme: Theme
}

class FileExtension extends React.PureComponent<FileExtensionProps> {

    render() {
        const styles = {
            backgroundColor: this.props.theme.pallete.surface.background,
            border: '1px solid ' + this.props.theme.pallete.divider,
            padding: '1rem',
        }

        return (
            <Text color='gray40' style={styles} weight='bold'>{this.props.extension}</Text>
        )
    }

}
