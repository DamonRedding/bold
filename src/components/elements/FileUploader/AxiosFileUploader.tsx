import axios, { AxiosRequestConfig, CancelTokenSource } from 'axios'
import React from 'react'

import { Omit } from '../../../util/types'

import { FileProps, FileUploader, FileUploaderProps } from './FileUploader'

export type UploadRequestType = (data: FormData, requestConfig: AxiosRequestConfig) => Promise<any>

export interface AxiosFileUploaderProps extends Omit<FileUploaderProps, 'file'> {
    uploadRequest: UploadRequestType
}

class AxiosFileUploader extends React.Component<AxiosFileUploaderProps, FileProps> {

    private cancelTokenSource: CancelTokenSource

    render() {
        return (
            <FileUploader
                file={this.state}
                onUpload={this.onUpload}
            />
        )
    }

    private onUpload = (selectedFile) => {
        const data = new FormData()
        data.append('file', selectedFile)

        this.cancelTokenSource = axios.CancelToken.source()

        this.setState({ selectedFile, uploading: true })

        const config: AxiosRequestConfig = {
            cancelToken: this.cancelTokenSource.token,
            onUploadProgress: (progressEvent: ProgressEvent) => {
                this.setState({ progress: Math.round((progressEvent.loaded * 100) / progressEvent.total) })
            },
        }

        this.props.uploadRequest(data, config).then((result) => {
            this.setState({ uploading: false })
        }).catch((error) => {
            if (axios.isCancel(error)) {
                this.setState({ error: false, selectedFile: undefined, uploading: false })
            } else {
                this.setState({ error: true, uploading: false })
            }
        })
    }

}
