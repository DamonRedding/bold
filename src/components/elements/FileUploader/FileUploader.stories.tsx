import { storiesOf } from '@storybook/react'
import * as React from 'react'

import { FileUploader } from './FileUploader'

storiesOf('Components/FileUploader', module)
    .add('initial', () => (
        <FileUploader
            text='Click or drop file here'
        />
    ))
    .add('uploading', () => (
        <FileUploader
            text='Click or drop file here'
            file={{
                progress: 40,
                selectedFile: { name: 'file_test.pdf', size: 4096 },
                uploading: true,
            }}
        />
    ))
    .add('completed', () => (
        <FileUploader
            text='Click or drop file here'
            file={{
                selectedFile: { name: 'file_test.pdf', size: 4096 },
            }}
        />
    ))
