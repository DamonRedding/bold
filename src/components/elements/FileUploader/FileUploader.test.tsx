import { mount, render } from 'enzyme'
import React from 'react'

import { withTheme } from '../../../test'

import { FileExtension, FileUploader } from './FileUploader'

const testFile: File = new File([], 'teste.pdf', { type: 'application/pdf' })
const testFile2: File = new File([], 'teste.pdf', { type: '' })

it('should render initial state', () => {
    expect(render(withTheme(<FileUploader />))).toMatchSnapshot()
})

it('should render uploading state', () => {
    expect(render(withTheme(
        <FileUploader
            file={{
                progress: 40,
                selectedFile: testFile,
                uploading: true,
            }}
        />
    ))).toMatchSnapshot()
})

it('should render complete state', () => {
    expect(render(withTheme(
        <FileUploader
            file={{
                selectedFile: testFile,
            }}
        />
    ))).toMatchSnapshot()
})

it('should render correct extension', () => {
    expect(mount(withTheme(
        <FileUploader
            file={{
                selectedFile: testFile,
            }}
        />
    )).find(FileExtension).props().extension
    ).toBe('pdf')
})

it('should render correct extension', () => {
    expect(mount(withTheme(
        <FileUploader
            file={{
                selectedFile: testFile2,
            }}
        />
    )).find(FileExtension).props().extension
    ).toBe(null)
})
