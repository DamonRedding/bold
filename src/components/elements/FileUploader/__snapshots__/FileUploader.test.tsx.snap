// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render complete state 1`] = `
.emotion-15 {
  background-color: #FFFFFF;
  border-radius: 4px;
  border: 1px solid #F2F2F7;
}

.emotion-6 {
  cursor: pointer;
  padding: 0.25rem;
}

.emotion-6 .accept-dashed-border {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid transparent;
  border-style: dashed;
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  margin-top: 0rem;
  margin-right: 0.25rem;
  margin-bottom: 0rem;
  margin-left: 0.25rem;
  display: inline-block;
}

.emotion-2:first-child {
  margin-left: 0;
}

.emotion-2:last-child {
  margin-right: 0;
}

.emotion-2:empty {
  display: none;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  fill: #53595E;
  font-size: 1.5rem;
}

.emotion-3 {
  color: #53595E;
  font-size: 0.875rem;
  font-weight: bold;
}

.emotion-14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 1px solid #F2F2F7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
}

.emotion-13 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 1rem;
}

.emotion-7 {
  color: #53595E;
  font-weight: bold;
  background-color: #F2F2F7;
  border: 1px solid #F2F2F7;
  padding: 1rem;
}

.emotion-12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-10 {
  font-size: 0.75rem;
  font-weight: bold;
  margin-right: 10px;
}

.emotion-11 {
  font-size: 0.75rem;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 5px;
}

.emotion-8 {
  fill: #056DFF;
  font-size: 1rem;
}

<div
  class="emotion-15"
>
  <div
    aria-disabled="false"
    class="emotion-6"
    style="position:relative"
  >
    <div
      class="accept-dashed-border"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-2"
        >
          <span
            class="emotion-1"
          >
            <svg
              aria-hidden="true"
              class="emotion-0"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M4 18h16v2H4zM11 7.94V16h2V7.94l3.95 4 1.41-1.42L12 4.11l-6.36 6.36 1.41 1.42zM12.06 7h-.12l.06-.06z"
              />
            </svg>
          </span>
        </div>
        <div
          class="emotion-2"
        >
          <span
            class="emotion-3"
          />
        </div>
      </div>
    </div>
    <input
      autocomplete="off"
      style="position:absolute;top:0;right:0;bottom:0;left:0;opacity:0.00001;pointer-events:none"
      type="file"
    />
  </div>
  <div
    class="emotion-14"
  >
    <span
      class="emotion-7"
    >
      PDF
    </span>
    <div
      class="emotion-13"
    >
      <div
        class="emotion-12"
      >
        <span
          class="emotion-9"
        >
          <svg
            aria-hidden="true"
            class="emotion-8"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M10.71 16.21l7-7-1.42-1.42-5.59 5.6-2.99-3-1.42 1.42"
            />
          </svg>
        </span>
        <span
          class="emotion-10"
        >
          file_test.pdf
        </span>
        <span
          class="emotion-11"
        >
          4 KB
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render initial state 1`] = `
.emotion-7 {
  background-color: #FFFFFF;
  border-radius: 4px;
  border: 1px solid #F2F2F7;
}

.emotion-6 {
  cursor: pointer;
  padding: 0.25rem;
}

.emotion-6 .accept-dashed-border {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid transparent;
  border-style: dashed;
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  margin-top: 0rem;
  margin-right: 0.25rem;
  margin-bottom: 0rem;
  margin-left: 0.25rem;
  display: inline-block;
}

.emotion-2:first-child {
  margin-left: 0;
}

.emotion-2:last-child {
  margin-right: 0;
}

.emotion-2:empty {
  display: none;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  fill: #53595E;
  font-size: 1.5rem;
}

.emotion-3 {
  color: #53595E;
  font-size: 0.875rem;
  font-weight: bold;
}

<div
  class="emotion-7"
>
  <div
    aria-disabled="false"
    class="emotion-6"
    style="position:relative"
  >
    <div
      class="accept-dashed-border"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-2"
        >
          <span
            class="emotion-1"
          >
            <svg
              aria-hidden="true"
              class="emotion-0"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M4 18h16v2H4zM11 7.94V16h2V7.94l3.95 4 1.41-1.42L12 4.11l-6.36 6.36 1.41 1.42zM12.06 7h-.12l.06-.06z"
              />
            </svg>
          </span>
        </div>
        <div
          class="emotion-2"
        >
          <span
            class="emotion-3"
          />
        </div>
      </div>
    </div>
    <input
      autocomplete="off"
      style="position:absolute;top:0;right:0;bottom:0;left:0;opacity:0.00001;pointer-events:none"
      type="file"
    />
  </div>
</div>
`;

exports[`should render uploading state 1`] = `
.emotion-14 {
  background-color: #FFFFFF;
  border-radius: 4px;
  border: 1px solid #F2F2F7;
}

.emotion-6 {
  cursor: pointer;
  padding: 0.25rem;
}

.emotion-6 .accept-dashed-border {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid transparent;
  border-style: dashed;
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  margin-top: 0rem;
  margin-right: 0.25rem;
  margin-bottom: 0rem;
  margin-left: 0.25rem;
  display: inline-block;
}

.emotion-2:first-child {
  margin-left: 0;
}

.emotion-2:last-child {
  margin-right: 0;
}

.emotion-2:empty {
  display: none;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  fill: #53595E;
  font-size: 1.5rem;
}

.emotion-3 {
  color: #53595E;
  font-size: 0.875rem;
  font-weight: bold;
}

.emotion-13 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 1px solid #F2F2F7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
}

.emotion-12 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 1rem;
}

.emotion-7 {
  color: #53595E;
  font-weight: bold;
  background-color: #F2F2F7;
  border: 1px solid #F2F2F7;
  padding: 1rem;
}

.emotion-10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 {
  font-size: 0.75rem;
  font-weight: bold;
  margin-right: 10px;
}

.emotion-9 {
  font-size: 0.75rem;
}

.emotion-11 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  height: 4px;
  width: 100%;
}

.emotion-11::-webkit-progress-bar {
  background-color: #F2F2F7;
  border-radius: 4px;
}

.emotion-11::-webkit-progress-value {
  background-color: #056DFF;
  border-radius: 4px;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.emotion-11::-moz-progress-bar {
  background-color: #056DFF;
  border-radius: 4px;
  -webkit-transition: all .3s;
  transition: all .3s;
}

<div
  class="emotion-14"
>
  <div
    aria-disabled="false"
    class="emotion-6"
    style="position:relative"
  >
    <div
      class="accept-dashed-border"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-2"
        >
          <span
            class="emotion-1"
          >
            <svg
              aria-hidden="true"
              class="emotion-0"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M4 18h16v2H4zM11 7.94V16h2V7.94l3.95 4 1.41-1.42L12 4.11l-6.36 6.36 1.41 1.42zM12.06 7h-.12l.06-.06z"
              />
            </svg>
          </span>
        </div>
        <div
          class="emotion-2"
        >
          <span
            class="emotion-3"
          />
        </div>
      </div>
    </div>
    <input
      autocomplete="off"
      style="position:absolute;top:0;right:0;bottom:0;left:0;opacity:0.00001;pointer-events:none"
      type="file"
    />
  </div>
  <div
    class="emotion-13"
  >
    <span
      class="emotion-7"
    >
      PDF
    </span>
    <div
      class="emotion-12"
    >
      <div
        class="emotion-10"
      >
        <span
          class="emotion-8"
        >
          file_test.pdf
        </span>
        <span
          class="emotion-9"
        >
          4 KB
        </span>
      </div>
      <progress
        class="emotion-11"
        max="100"
        value="40"
      />
    </div>
  </div>
</div>
`;
